commons-geometry-core/src/main/java/org/apache/commons/geometry/core/partitioning/AbstractConvexHyperplaneBoundedRegion.java

```diff
@@ -161,7 +161,7 @@ public String toString() {
      */
     protected <R extends AbstractConvexHyperplaneBoundedRegion<P, S>> R transformInternal(
             final Transform<P> transform, final R thisInstance, final Class<S> boundaryType,
-            final Function<List<S>, R> factory) {
+            final Function<? super List<S>, R> factory) {
```

Comment:
I see the reasoning behind this but I'm not sure if it's needed since tt makes an already crazy method signature even crazier. WDYT? This goes for the other `super` generic changes as well.

Reply:
IMO, it improves the usability of the APIs. See Josh Bloch's _Effective Java_ (3rd Edition):

> Using wildcard types in your APIs, while tricky, makes the APIs far more flexible. If you write a library that will be widely used, the proper use of wildcard types should be considered mandatory.

Actually, some of the generic parameters can also wildcards (i.e. `final Class<? extends S> boundaryType`), but I kept those out for now to avoid overcomplicating the signature even more.

Reply:
Okey dokey.